test/component.test.js

```diff
@@ -156,6 +156,22 @@ Repeat Header`;
       assume(tree.find('#blort')).to.have.length(0);
     });
 
+
+    it('Headers do not increment on update', () => {
+      const input = `
+# Header
+# Header
+`;
+
+      renderFullDom({ source: input });
+      assume(tree.find('#header')).to.have.length(1);
+      assume(tree.find('#header-1')).to.have.length(1);
+
+      tree.update();
+
+      assume(tree.find('#header-2')).to.have.length(0);
+    });
```

Comment:
I may be misunderstanding the test, but I am fairly certain this should replicate the issue. Any ideas @clinejm @dustinsavery